Userland/Libraries/LibGfx/ImageFormats/BilevelImage.cpp

```diff
@@ -58,21 +58,20 @@ void BilevelImage::fill(bool b)
         byte = fill_byte;
 }
 
-void BilevelImage::blend_into(BilevelImage& out, IntPoint position, BlendType operator_) const
+template<BilevelImage::BlendType operator_>
+void BilevelImage::blend_into(BilevelImage& out, IntPoint position) const
 {
-    static constexpr auto combine = [](bool dst, bool src, BlendType op) -> bool {
-        switch (op) {
-        case BlendType::Or:
+    constexpr auto combine = [](bool dst, bool src) -> bool {
+        if constexpr (operator_ == BlendType::Or)
             return dst || src;
-        case BlendType::And:
+        if constexpr (operator_ == BlendType::And)
```

Comment:
is moving away from `switch` necessary, or do you still get the same speedup by keeping switch and moving to passing in the operator as template param? I would expect the latter, and that way you'll keep getting compiler diags when the number of enum entries change

Reply:
You'll get a compile error in the non-template overload by default, and an error here when you add the new value in the non-template overload. So that should be fine.

Didn't test it, but it should only be a difference at `O0` where I expect the `constexpr` version to be optimized but not the `switch`. At higher optimization level, it doesn't make a difference.

Reply:
Can we keep the LHS then?